File: grading/grading.clj
(defn get-next-multiple [denominator integer]
  (let [curr-int (inc integer)]
    (if (= 0 (mod curr-int denominator))
      curr-int
      (get-next-multiple denominator curr-int))))

(defn gradingStudents [grades]
  (map
   (fn [grade]
     (let [next-five-multiple (get-next-multiple 5 grade)
           should-ret-grade? (or (< grade 38)
                                 (>= (- next-five-multiple grade) 3))]
       (if should-ret-grade? grade next-five-multiple)))
   grades))

File: grading/grading.js
// https://www.hackerrank.com/challenges/grading/problem

const getNextMultiple = (denominator, integer) => {
  let currInt = integer + 1;

  return currInt % denominator === 0
    ? currInt
    : getNextMultiple(denominator, currInt);
};

const gradingStudents = grades =>
  grades.map(grade => {
    const nextFiveMultiple = getNextMultiple(5, grade);

    const shouldRetGrade = grade < 38 || nextFiveMultiple - grade >= 3;

    return shouldRetGrade ? grade : nextFiveMultiple;
  });
